### PR TITLE
 Fix Form Changes with Secondary Fusion Component

### PR DESCRIPTION
**Pull request by bennybroseph**
Allows form changes when you have a Pokemon chosen as the second component of a fusion. i.e. Charizard/Snorlax can still be Dynamaxed with a Max Mushroom.

This also allows these items to spawn where they wouldn't previously.

https://github.com/pagefaultgames/pokerogue/assets/13838608/7e2bb4f5-d1c2-4910-9a36-7cf506893696

### CODE REVIEW

**Comment by bennybroseph** [comment]
Regular form change for sanity
https://github.com/pagefaultgames/pokerogue/assets/13838608/6b8c0687-aee9-4fa0-ba9c-8e729af70390

File: src/field/pokemon.ts [hunk 1]
```diff
@@ -2507,8 +2507,15 @@ export class PlayerPokemon extends Pokemon {
   }
 
   changeForm(formChange: SpeciesFormChange): Promise<void> {
```
Comment (Xavion3) [inline src/field/pokemon.ts:2509]: There's two changeForm functions, why aren't you doing anything with the second one?

**Review by Xavion3 — CHANGES_REQUESTED** [review]
Nice, I was working on basically this feature myself too, so I've got some grounding for it.

How does this handle form changes which match both halves? That was one of the trickiest things to me, supporting independent form changes.

File: src/field/pokemon.ts (hunk 1 above)
Comment (bennybroseph) [reply src/field/pokemon.ts:2510]: Ah, both very good points. I haven't tested a double fusion like that before. Does seems tricky, but also probably fine to include without support for now right? I imagine most people are fusing two different things and enabling that to work is better than waiting for an edge case fix.

As far as the other changeForm function, it must not have gotten called on anything I was testing so I didn't see I needed to make changes. Taking a glance, it looks like it's nearly identical to the other one so it won't take much to update it. I'm now curious why it exists though and when it gets called.

I'll take a look at both of these later today.

Comment (bennybroseph) [reply src/field/pokemon.ts:2517]: I've updated the second function as requested. Still not sure when it gets called, but it matches the functionality of the other one now. 

As for double fusions, do you think that should be a separate PR?

**Comment by bennybroseph** [comment]
This is probably doable now but will need to be nearly restarted. Closing this out for now.

**Pull request closed without merging**